lib/grpnames.gi

```diff
@@ -81,16 +81,12 @@ InstallMethod( IsTrivialNormalIntersection,
 InstallGlobalFunction( UnionIfCanEasilySortElements,
 
   function( arg )
-    local i, N;
 
-    for i in [1..Length(arg)] do
-      for N in arg[i] do
-        if not CanEasilySortElements(N) then
-          return Concatenation(arg);
-        fi;
-      od;
-    od;
-    return Union(arg);
+    if ForAll(arg, CanEasilySortElements) then
```

Comment:
The code as previously written was literally:
```suggestion
    if ForAll(arg, N -> ForAll(N, CanEasilySortElements)) then
```
Is what you've written equivalent to this? Just wanted to check. (Sorry - I'm ignorant of the details of `CanEasilySortElements`).

Reply:
That's a very sensible question, of course! So, there is this method:
```gap
InstallMethod(CanEasilySortElements,"homogeneous list",
  [IsHomogeneousList],
function(l)
  return Length(l)=0 or CanEasilySortElements(l[1]);
end);
```
So my version should be fine, as long as all list over which we take the union are homogenous, which they are in the usescases we apply it. *But* thinking about this again, of course `UnionIfCanEasilySortElements`, while undocumented, could in principle be called of any kind of input, so perhaps we should be defensive here?

OTOH, the worst that can go wrong is that we take a concatenation when we could have perform a union. Also, `AddSetIfCanEasilySortElements(list, obj)` already seems to rely on this behavior, as it calls `CanEasilySortElements(list)` (and also not `CanEasilySortElements(obj)`)